### DISCUSSION TITLE
CRA Build gave me error BigInt

### DISCUSSION THREAD

**Discussion by ombayoeco11** [wevm/viem#621] [Question]
Before wagmi got viem, everythings fine. After the update, everytime i build my codes. i always got error like this
ContractFunctionExecutionError: An unknown error occurred while executing the contract function "allowance".

Raw Call Arguments:
  to:    0x337610d27c682E347C9cD60BD4b3b107C9d34dDd
  data:  0xdd62ed3e00000000000000000000000027c2f3689d218837baee2dd72fc6695e997777560000000000000000000000002bf8e1edf35d20cfe7918b3e2da59cf619f4504c
 
Contract Call:
  address:   0x337610d27c682E347C9cD60BD4b3b107C9d34dDd
  function:  allowance(address owner, address spender)
  args:               (0x27C2F3689d218837BAee2dd72FC6695e99777756, 0x2BF8E1edF35D20cFe7918B3e2da59cF619f4504C)

Docs: https://viem.sh/docs/contract/readContract.html
Details: Cannot convert a BigInt value to a number
Version: viem@0.3.37
    at Ys (getContractError.ts:55:10)
    at readContract.ts:100:11
    at f (regeneratorRuntime.js:44:17)
    at Generator.<anonymous> (regeneratorRuntime.js:125:22)
    at Generator.throw (regeneratorRuntime.js:69:21)
    at r (asyncToGenerator.js:3:20)
    at l (asyncToGenerator.js:25:9)

while my code is this 
 const checkAllowance = async () => {
      try {

        // Call the `allowance` function on the token contract

        // Convert the result to human-readable format
        const allowing = ethers.utils.formatUnits(allowance, 18);
        setAllow(allowing);
        setUsdt()
         console.log("Allowance USDT :", allow, allowance);
         const result = await readContract({
           address: USDTBEP20,
           abi: USDTBEP20ABI,
           functionName: "allowance",
           args: [account.toString(), NFTADDR],
         });
         if (result) {
           const allowing = ethers.utils.formatUnits(BigNumber.from(result), 18);
           setAllow(allowing);
           setUsdt(balances.usdt);
         } else {
           setAllow("0");
         }
      } catch (error) {
        setError(error.reason);
        console.log(error);
      }
    };

anyone can explain to me whats wrong?

**Comment by N3ST3DLabs** [comment]
I getting a reply "On"  instead of a uint 256 

`  const { data: allowance, refetch } = useContractRead({
    address: "0x7c3118d6D468A1a7386F86c50Bc00F0f40F7E1c0",
    abi: ERC20ABI,
    functionName: "allowance",
    args: [user.user.address, "0x1111111254EEB25477B68fb85Ed929f73A960582"],
  });`
